pytorch_lightning/trainer/trainer.py

```diff
@@ -1110,10 +1110,14 @@ def _call_teardown_hook(self, model: 'pl.LightningModule') -> None:
 
         if self.datamodule is not None:
             self.datamodule.teardown(stage=fn)
-        self.profiler.teardown(stage=fn)
+
         self.teardown(stage=fn)
         model.teardown(stage=fn)
```

Comment:
I'm moving `self.profiler.teardown` to be after `self.teardown`, as otherwise we miss the `teardown` hook itself being profiled.